flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOptions.java

```diff
@@ -47,9 +48,10 @@
 	 * Choice of timer service implementation.
 	 */
 	@Documentation.Section(Documentation.Sections.STATE_BACKEND_ROCKSDB)
-	public static final ConfigOption<String> TIMER_SERVICE_FACTORY = ConfigOptions
+	public static final ConfigOption<PriorityQueueStateType> TIMER_SERVICE_FACTORY = ConfigOptions
```

Comment:
Is it a problem that the type of the config option changes. You mentioned to me before that this could impact users, which is correct.

I would not be too worried, though, because most users probably would not use this in code but rather in the yaml config.

Reply:
Also, weirdly named option: it's a "factory" but the type is an enum? 😅

Reply:
> I would not be too worried, though, because most users probably would not use this in code but rather in the yaml config.

That was also my thinking after looking at it again. The class `RocksDBOptions` has no annotation, so it is not necessarily `Public`. Moreover I think we do not advertise using those options directly. Programatically users should rather use `org.apache.flink.contrib.streaming.state.RocksDBStateBackend#setPriorityQueueStateType`, if I am not mistaken.